python/ray/data/read_api.py

```diff
@@ -1654,13 +1663,31 @@ def read_tfrecords(
             By default, the number of output blocks is dynamically decided based on
             input data size and available resources. You shouldn't manually set this
             value in most cases.
-
+        tfx_read_options: Specifies read options when reading TFRecord files with TFX.
+            When no options are provided, the default version without tfx-bsl will
+            be used to read the tfrecords.
     Returns:
         A :class:`~ray.data.Dataset` that contains the example features.
 
     Raises:
         ValueError: If a file contains a message that isn't a ``tf.train.Example``.
     """
+    import platform
+
+    tfx_read = False
+
+    if tfx_read_options and platform.processor() != "arm":
+        try:
+            import tfx_bsl  # noqa: F401
+
+            tfx_read = True
+        except ModuleNotFoundError:
+            logger.warning(
+                "Please install tfx-bsl package with"
+                " `pip install tfx_bsl --no-dependencies`."
+                " This can help speed up the reading of large TFRecord files."
+            )
```

Comment:
when user has specified `tfx_read_options` but `tfx_bsl` isn't installed, it'd be better to just throw an exception. 
And I guess it's okay to not check platform here, as tfx_bsl may support arm in the future.

Reply:
it does not support it right now, I find it a lot safer to not assume it will

Reply:
> when user has specified tfx_read_options but tfx_bsl isn't installed, it'd be better to just throw an exception.

So you are suggestion I just remove the whole block of trying to import and logging a warning?